packages/design-system/src/components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,280 @@
+import Ellipses from './Ellipses'
+import Page from './Page'
+import React from 'react'
+import classNames from 'classnames';
+
+export interface PaginationProps {
+  /**
+   * Class to be applied to parent `<nav>` element of Pagination component. Optional.
+   */
+  className?: string,
+  /**
+   * Renders compact layout. Optional.
+   */
+  compact?: boolean,
+  /**
+   * Defines active page in Pagination. Optional. 
+   */
+  currentPage: number,
+  /**
+   * Sets a custom url for Pagination links. Optional.
+   */
+  customUrl?: string,
+  /**
+   * A callback function used to handle state changes.
+   */
+  onPageChange: (evt: React.MouseEvent, page: number) => void,
+  /**
+   * Sets custom label on left navigation. Optional.
+   */
+  leftLabel?: string,
+  /**
+   * Sets custom label on right navigation. Optional.
+   */
+  rightLabel?: string,
+  /**
+   * Sets total number of pages in Pagination component.
+   */
+  totalPages: number,
+}
+
+// Determines number of pages visible to either side of active page.
+const overflow = 1;
+
+// Determines total number of visible pages without Ellipses.
+const maxVisiblePages = 7;
+
+function paginationBuilder(page: number, pages: number):number[] {
+  const paginationRange = [];
+
+  let start = page - overflow;
+  let end = page + overflow;
+  
+  const availableSlots = maxVisiblePages - 2;
+  
+  /**
+   * If the current page is < `maxVisiblePages`, 
+   * add 1 - 5 pages.
+   */
+  if (page < availableSlots) {
+    start = 1;
+    end = availableSlots;
+  }
+
+  /**
+   * If the current page equals `pages` - 1, 
+   * make sure `start` begins one page earlier.
+   */
+  if (page === pages - 2) {
+    start -= 1;
+    end += 1;
+  }
+
+  /**
+   * If `end` page is two from the end, 
+   * make sure the last page shows instead of ellipsis.
+   */
+  if (end === pages - 2) {
+    end += 1;
+  }
+
+  /**
+   * If `end` > `pages`, 
+   * add last pages to `paginationRange[]`.
+   */
+  if (end >= pages) {
+    start = pages - (availableSlots - 1); 
+    end = pages;
+  }
+
+  /**
+   * If `pages` is 5 or fewer, 
+   * all pages added to `paginationRange[]`
+   */
+  if (pages <= maxVisiblePages) {
+    start = 1;
+    end = pages;
+  }
+
+  for (let i = start; i <= end; i++) {
+    paginationRange.push(i);
+  }
+  
+  return paginationRange;
+}
+
+function Pagination({className, compact, currentPage, customUrl, onPageChange, leftLabel, rightLabel, totalPages}: PaginationProps): React.ReactElement {
+  const pageChange = React.useCallback(page => (evt: React.MouseEvent) => onPageChange(evt, page), [onPageChange])
+  
+  const pageRange = paginationBuilder(currentPage, totalPages)
```

Comment:
[question] if the component is in mobile or compact mode is it worth skipping this paginationBuilder calculation since that info is not used?

I also understand that the calculation isn't very intensive, but just an idea.